flat_mutation_reader.cc

```diff
@@ -242,6 +242,13 @@ flat_mutation_reader make_reversing_reader(flat_mutation_reader original, query:
     return make_flat_mutation_reader<partition_reversing_mutation_reader>(std::move(original), max_size);
 }
 
+flat_mutation_reader maybe_reverse(flat_mutation_reader r, const query::partition_slice& s, query::max_result_size max_size) {
+    if (s.options.contains(query::partition_slice::option::reversed)) {
+        return make_reversing_reader(std::move(r), std::move(max_size));
+    }
+    return r;
```

Comment:
This patch deserves special treatment too, but I'd like @denesb ack before.

Reply:
@avikivity What special treatment are you talking about?

Reply:
I previous iteration had a patch which I queued, and now this iteration is rebased over it so you can't see it. The special treatment is queuing it ahead of the series in order to unblock others (since it's not restricted to mx reader). We can keep it in the series of course.